core-bundle/src/Controller/ContentElement/AbstractContentElementController.php

```diff
@@ -22,6 +22,13 @@ abstract class AbstractContentElementController extends AbstractFragmentControll
 {
     public function __invoke(Request $request, ContentModel $model, string $section, array $classes = null): Response
     {
+        if (!$this->isVisible($model, $request)) {
```

Comment:
Good plan!
But unfortunately you will not get there because the element will be never loaded previously because of `ContentModel::findPublishedByPidAndTable` - see for example in the article (there are some other places!): https://github.com/contao/contao/blob/37e1a09d99f6bbde827d1e4e7f47ed41341d6bcb/core-bundle/src/Resources/contao/modules/ModuleArticle.php#L189

Especially for a start date in the future we need to find another solution! Because otherwise the shared max age will be never added to the response, so the page cache won't be invalid on the content element start date!

Reply:
I guess for BC reasons, we need to make a difference between old style elements/modules and new fragments.
New fragments should always be loaded and the response should be generated. In case of an invisible one, it should return an empty response. This would allow to also modify the caching headers for `start` and `stop`.
I'll mark this issue as up for discussion.